theano/tensor/opt.py

```diff
@@ -4883,14 +4883,25 @@ def local_useless_elemwise_comparison(node):
         return
     if node.op.scalar_op.nin != 2:
         return
+
+    def zeros_like(model, dtype):
+        ret = T.zeros_like(node.inputs[0], dtype=node.outputs[0].dtype)
```

Comment:
this introduce a but. It should be model and not node.inputs[0]. I did a PR to fix it:

https://github.com/Theano/Theano/pull/4891